libdnf/dnf-sack.cpp

```diff
@@ -224,17 +224,31 @@ dnf_sack_new(void)
     return DNF_SACK(g_object_new(DNF_TYPE_SACK, NULL));
 }
 
-static int
-can_use_repomd_cache(FILE *fp_solv, unsigned char cs_repomd[CHKSUM_BYTES])
-{
-    unsigned char cs_cache[CHKSUM_BYTES];
-
-    if (fp_solv &&
-        !checksum_read(cs_cache, fp_solv) &&
-        !checksum_cmp(cs_cache, cs_repomd))
-        return 1;
+// Try to load cached solv file otherwise return FALSE
+static gboolean
+try_to_use_cached_solvfile(const char *path, Repo *repo, int flags, const unsigned char *checksum, GError **err){
+    FILE *fp_cache = fopen(path, "r");
+    if (!fp_cache) {
+        return FALSE;
+    }
```

Comment:
If the opening fails we have no way to know. dnf will keep working, just without cache and that's not great, we should at least log the error here (again on a higher than debug level).

Reply:
That seems like a good change, I have added that. :+1: